imageio/imageio-tiff/src/test/java/com/twelvemonkeys/imageio/plugins/tiff/TIFFImageReaderTest.java

```diff
@@ -240,6 +242,28 @@ public void testReadOldStyleJPEGGrayscale() throws IOException {
         }
     }
 
+    @Test
+    public void testReadGeoTIFFFloat32() throws IOException {
+        TestData testData = new TestData(getClassLoaderResource("/tiff/geotiff-float32.tif"), new Dimension(299, 221));
+
+        try (ImageInputStream stream = testData.getInputStream()) {
+            TIFFImageReader reader = createReader();
+            reader.setInput(stream);
+
+            IIOReadWarningListener warningListener = mock(IIOReadWarningListener.class);
+            reader.addIIOReadWarningListener(warningListener);
+
+            BufferedImage image = reader.read(0);
```

Comment:
This line has to be replaced with:

    Raster raster = reader.readRaster(0, null);

as a `BufferedImage` need float samples normalised to the range [0...1].